src/interface/blas3/backend/nvidia_gpu.hpp

```diff
@@ -53,31 +54,70 @@ typename executor_t::policy_t::event_t _gemm(
                                                                     _ldc,
                                                                     batch_size);
   }
-  if (_M < 512 && _N < 512 && _K < 512) {
+
+#ifdef SB_ENABLE_JOINT_MATRIX
+  const char* en_joint_matrix = std::getenv("SB_ENABLE_JOINT_MATRIX");
```

Comment:
I can see you've described this in the PR description. Why do we want users to set an environment variable? That's not very intuitive and not something we've done in SYCL-BLAS before. Shouldn't we use joint matrix whenever we detect that it is supported and that we support the sizes?

Reply:
I would be a good idea introducing a CMake flag similar to the existing [ones](https://github.com/codeplaysoftware/sycl-blas/blob/master/doc/Gemm.md#relevant-cmake-variables) for enabling the joint matrices.

In any case, we could adopt an approach similar to NVIDIA by having an [environment variable](https://docs.nvidia.com/cuda/cutensor/user_guide.html#environment-variables) for disabling the TF32, such as `NVIDIA_TF32_OVERRIDE=0`.

Reply:
@mehdi-goli suggested that we should compile all supported kernels and then let the selection of the appropriate kernel depend on the environment variable. This is inline with what NVIDIA does with their DNN and BLAS libraries where they use the `NVIDIA_TF32_OVERRIDE` environment variable to select whether Tensorcores get invoked or not. So we are trying to mimic the same in our libraries for the `joint_matrix` based implementations.

Reply:
Right, that's a weird design but I'll resolve for now.